doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
 
 For patch releases we never allow changes in the public API, unless the change is specifically to fix a bug that aligns the actual behavior of the code with the publicly documented behavior.
 
+Only release branches (more on those later) include the full major.minor.patch
+version numbers included with releases. The major and minor version of the _master_ and
+_develop_ branches do not include a patch number and always have a SNAPSHOT tag.
+The _master_ and _develop_ versions always matches the pattern *MAJOR*.*MINOR*-SNAPSHOT.
+
+The _master_ version always has the same major and minor number as the latest
+release while the _develop_ version uses the next expected major/minor
+release number, such as 2.1-SNAPSHOT and 2.2-SNAPSHOT respectively
+(after a 2.1.x release, but before any 2.2.x releases are made public).
+
 == Workflow
 
-RDF4J uses a git branching model where feature development takes place on branches from the master branch. This is where all development for the next (minor or major) release happens.
+RDF4J uses a git branching model where feature development takes place on branches from the _develop_ branch. This is where all development for the next (minor or major) release happens.
+These branches should have the prefix `features/` to indicate they are either
+not complete or have not been scheduled for the next minor release.
 
 Every issue, no matter how small, gets its own branch while under development. Issue branch names are always prefixed with `issues/`, followed by the issue number in the GitHub issue tracker, followed by one or two dash-separated keywords for the issue.
 
 For example: `issues/#276-rdfxml-sax` is the branch for a fix for GitHub issue https://github.com/eclipse/rdf4j/issues/276[#276], which has to do with RDF/XML and the SAX parser.
 
-Once a feature/fix is complete, tested, and reviewed (via a Pull Request), it is merged into master, and the issue branch is closed.
+Once a issue is complete and tested, a Pull Request should be created for
+peer review.
+
+=== Patch Requests
+
+If the change is a bug fix, contains no new features, and
+does not change any public or protected APIs (class or method signatures in the
+core module, i.e. code included in the release archive) then
```

Comment:
I don't quite understand what the reference to core and release archive is here. I would prefer that we define the public API and refer to it instead of adding novel terminology.